csi/controller.go

```diff
@@ -1128,6 +1128,32 @@ func (s *OsdCsiServer) ListSnapshots(
 	ctx context.Context,
 	req *csi.ListSnapshotsRequest,
 ) (*csi.ListSnapshotsResponse, error) {
+	// Get any labels passed in by the CO
+
+	_, ok := req.GetSecrets()[osdSnapshotCredentialIDKey]
```

Comment:
It is possible to pass this value manually via Nomad:
https://developer.hashicorp.com/nomad/docs/commands/volume/snapshot-list

With k8s, I check and this should still work. CSI will find the snapshot class for a given snapshot, and pass the associated secrets into our ListSnapshots call:
https://github.com/kubernetes-csi/external-snapshotter/blob/master/pkg/sidecar-controller/snapshot_controller.go#L258-L266